src/_mixins-functions.scss

```diff
@@ -60,55 +60,97 @@
-/// https://gist.github.com/marcedwards/3446599.
+/// Easily apply a HiDPI background image. Autoprefixer will take care of
+/// generating the non-standard `-webkit-min-device-pixel-ratio` for
+/// browsers/devices that need it.
 ///
-/// N.B. Autoprefixer will take care of generating the non-standard
-/// `-webkit-device-pixel-ratio` for browsers/devices that need it, see:
-/// http://caniuse.com/#feat=css-media-resolution.
+/// @link http://caniuse.com/#feat=css-media-resolution - Browser support.
 ///
-/// @author Chris Pearce
-///
-/// @access public
+/// @param {String} $file - The image filename and path if required.
+/// @param {Number} $width [auto] - The image width + its unit e.g. `px`, or
+/// the `em()` or `rem()` functions can be used.
+/// @param {Number} $height [$width] - Same as width but for the height and
+/// the height is optional, if empty it'll default to `$width`.
 ///
-/// @param {String} $img-url — The image url path
-/// @param {Number [unit]} $img-width [auto] — The image width in `px`
-/// @param {Number [unit]} $img-height [auto] — The image height in `px`
+/// @throws '`$file` must be of type "string"'
+/// @throws '`$width` must be of type "number"'
+/// @throws '`$width` must have a unit'
+/// @throws '`$height` must be of type "number"'
+/// @throws '`$height` must have a unit'
+/// @throws 'The `$shell-g-hidpi-mq` setting is required'
 ///
 /// @example scss - Usage
 ///     .element-1 {
-///         @include hidpi-bg-img("path/to/image/image.png", 100px, 45px);
+///         @include hidpi-bg-img('img/icons/tick@2x.png', 100px, 45px);
 ///     }
 ///     .element-2 {
-///         @include hidpi-bg-img("path/to/image/image.png", 32px);
+///         @include hidpi-bg-img('arrow@2x.png', 32px);
+///     }
+///     .element-3 {
+///         @include hidpi-bg-img('delete@2x.png', rem(134), rem(65));
 ///     }
+///
 /// @example css - CSS Output
 ///     .element-1 {
-///         @media (min-resolution: 1.3dppx),
-///                (min-resolution: 124.8dpi) {
-///             background-image: url("path/to/image/image.png");
+///         @media (min-resolution: 2dppx),
+///                (min-resolution: 192dpi) {
+///             background-image: url('img/icons/tick@2x.png');
 ///             background-size: 100px 45px;
 ///         }
 ///     }
 ///     .element-2 {
-///         @media (min-resolution: 1.3dppx),
-///                (min-resolution: 124.8dpi) {
-///             background-image: url("path/to/image/image.png");
+///         @media (min-resolution: 2dppx),
+///                (min-resolution: 192dpi) {
+///             background-image: url('arrow@2x.png');
 ///             background-size: 32px 32px;
 ///         }
 ///     }
+///     .element-3 {
+///         @media (min-resolution: 2dppx),
+///                (min-resolution: 192dpi) {
+///             background-image: url('delete@2x.png');
+///             background-size: 8.375rem 4.062rem;
```

Comment:
I see what's happening, so the actual implementation does use whole numbers (see the `@example scss - Usage` section above):

```scss
@include hidpi-bg-img('delete@2x.png', rem(134), rem(65));
```

This is the output:

```css
background-size: 8.375rem 4.062rem;
```